storage/series/series.go

```diff
@@ -514,12 +585,35 @@ func (s *dbSeries) Close() {
 	// NB(r): We explicitly do not place this ID back into an
 	// existing pool as high frequency users of series IDs such
 	// as the commit log need to use the reference without the
-	// overhead of ownership tracking.
+	// overhead of ownership tracking. In addition, the blocks
+	// themselves have a reference to the ID which is required
+	// for the LRU/WiredList caching strategy eviction process.
+	// Since the wired list can still have a reference to a
+	// DatabaseBlock for which the corresponding DatabaseSeries
+	// has been closed, its important that the ID itself is still
+	// available because the process of kicking a DatabaseBlock
+	// out of the WiredList requires the ID.
+	//
 	// Since series are purged so infrequently the overhead
 	// of not releasing back an ID to a pool is amortized over
 	// a long period of time.
 	s.id = nil
 
+	switch s.opts.CachePolicy() {
+	case CacheLRU:
+		// In the CacheLRU case, blocks that were retrieved from disk are owned
+		// by the WiredList and should not be closed here. They will eventually
+		// be evicted and closed by  the WiredList when it needs to make room
+		// for new blocks.
+		for _, block := range s.blocks.AllBlocks() {
+			if !block.WasRetrievedFromDisk() {
+				block.Close()
+			}
+		}
+	default:
+		s.blocks.RemoveAll()
```

Comment:
Is this necessary if `blocks.Reset()` is about to get called?

Reply:
yep cause Reset doesn't explicitly close the underlying blocks